From 0e155a4ea8951ee677d0a79f2a72ea5e23f27fe2 Mon Sep 17 00:00:00 2001
From: Divya Ranjan <divya@subvertising.org>
Date: Wed, 11 Dec 2024 01:15:01 +0000
Subject: [PATCH 08/10] gnu: Add rust-gtk4-macros-0.4.

* gnu/packages/crates-gtk.scm (rust-gtk4-macros-0.4): New variable.

Change-Id: Ie9d788cc69c818dd72e6248107882468af8d9860
---
 gnu/packages/crates-gtk.scm | 31 +++++++++++++++++++++++++++++++
 1 file changed, 31 insertions(+)

diff --git a/gnu/packages/crates-gtk.scm b/gnu/packages/crates-gtk.scm
index 3646bc9a8c..7272c55689 100644
--- a/gnu/packages/crates-gtk.scm
+++ b/gnu/packages/crates-gtk.scm
@@ -4277,6 +4277,37 @@ (define-public rust-gtk4-macros-0.7
                                    ("rust-gtk4" ,rust-gtk4-0.7)
                                    ("rust-trybuild2" ,rust-trybuild2-1))))))
 
+(define-public rust-gtk4-macros-0.4
+  (package
+    (name "rust-gtk4-macros")
+    (version "0.4.10")
+    (source
+      (origin
+        (method url-fetch)
+        (uri (crate-uri "gtk4-macros" version))
+        (file-name
+          (string-append name "-" version ".tar.gz"))
+        (sha256
+          (base32
+            "0v42i8xpg9f84iq1d0k2sb7vh94n9v9rk7i7iq3579wi9ra0pfka"))))
+    (build-system cargo-build-system)
+    (arguments
+      `(#:cargo-inputs
+        (("rust-anyhow" ,rust-anyhow-1)
+         ("rust-proc-macro-crate"
+          ,rust-proc-macro-crate-1)
+         ("rust-proc-macro-error"
+          ,rust-proc-macro-error-1)
+         ("rust-proc-macro2" ,rust-proc-macro2-1)
+         ("rust-quick-xml" ,rust-quick-xml-0.22)
+         ("rust-quote" ,rust-quote-1)
+         ("rust-syn" ,rust-syn-1))))
+    (home-page "https://gtk-rs.org/gtk4-rs")
+    (synopsis "Macros helpers for GTK 4 bindings")
+    (description
+      "This package provides Macros helpers for GTK 4 bindings.")
+    (license license:expat)))
+
 (define-public rust-gtk4-sys-0.8
   (package
     (name "rust-gtk4-sys")
-- 
2.46.0

